src/main/resources/default/taglib/w/multiSelect.html.pasta

```diff
@@ -1,45 +1,204 @@
-<i:arg name="maxItems" type="int" default="0" />
-<i:arg name="required" type="boolean" default="false" />
-<i:arg name="readonly" type="boolean" default="false" />
-<i:arg name="adminOnly" type="boolean" default="false" />
-<i:arg name="forceFullWidth" type="boolean" default="false" />
-<i:arg name="allowCustomEntries" type="boolean" default="false" />
-
-<i:pragma name="inline" value="true" />
-<i:pragma name="description" value="Renders a mutliselect dropdown field within a Wondergem template" />
-
-<div class="col-xs-@smallSpan col-md-@span form-group @user.signalFieldError(name)">
-    <i:if test="isFilled(label)">
-        <label>
-            <span class="@if (required) { input-required } @if (adminOnly) { admin-link }">
-                @label
-            </span>
-        </label>
-    </i:if>
-
-    <select name="@name" class="select2-select form-control @user.signalFieldError(name)"
-            @if (allowCustomEntries) { data-tags="true" }
-            @if (maxItems != 0) { data-maximum-selection-length="@maxItems" }
-            @if(forceFullWidth) { style="width: 100%;" }
-            multiple="multiple">
-        <i:render name="body" />
-    </select>
-
-    <i:if test="isFilled(help)">
-        <span class="help-block"><i:raw>@help</i:raw></span>
-    </i:if>
-    <i:if test="user.hasError(name)">
-        <span class="error-block help-block">@user.getFieldErrorMessage(name)</span>
-    </i:if>
-    <i:if test="optional">
-        <input name="@name" type="hidden" value="" />
-    </i:if>
+<i:arg name="span" type="int" default="6"/>
+<i:arg name="smallSpan" type="int" default="12" description="Defines the span for mobile devices"/>
+<i:arg name="name" type="String"/>
+<i:arg name="labelKey" type="String" default=""/>
+<i:arg name="label" type="String" default="@i18n(labelKey)"/>
+<i:arg name="helpKey" type="String" default=""/>
+<i:arg name="help" type="String" default="@i18n(helpKey)"/>
+<i:arg name="optional" type="boolean" default="false"/>
+<i:arg name="maxItems" type="int" default="0" description="Maximum number of items allowed in the field. 0 = unlimited."/>
+<i:arg name="required" type="boolean" default="false"/>
+<i:arg name="readonly" type="boolean" default="false"/>
+<i:arg name="adminOnly" type="boolean" default="false"/>
+<i:arg name="allowCustomEntries" type="boolean" default="false"/>
+
+<i:local name="id" value="@apply('multiselect-%s', call.generateLocalId())"/>
+
+<div id="@id" class="multiselect">
+    <w:textfield id="@apply('%s-input',id)" name="@apply('%s-input', name)" span="@span" smallSpan="@smallSpan"
+                 value="" label="@label" help="@help" adminOnly="@adminOnly" readonly="@readonly" required="@required"/>
 </div>
+
+<select id="@id-suggestions-select" style="display: none" multiple="multiple">
+    <i:render name="body"/>
+</select>
+
+<i:if test="optional">
+    <input name="@name" type="hidden" value=""/>
+</i:if>
+
+<script>
+    var initMultiSelect = function (args) {
```

Comment:
I could move that in a separate JS file, any idea where?

Reply:
Maybe `page-script.html.pasta`?

Reply:
yep